yt/yt/server/lib/exec_node/config.h

```diff
@@ -664,11 +664,40 @@ DEFINE_REFCOUNTED_TYPE(TNbdConfig)
 
 ////////////////////////////////////////////////////////////////////////////////
 
+DEFINE_ENUM(EJobProxyLoggingMode,
+    (Sandbox)
+    (SeparateDirectory)
+);
+
+////////////////////////////////////////////////////////////////////////////////
+
+class TJobProxyLoggingEnvironment : public NYTree::TYsonStruct {
+public:
+    EJobProxyLoggingMode LoggingMode;
+
+    std::optional<TString> LoggingDirectory;
+
+    NLogging::TLogManagerConfigPtr LoggingTemplate;
+
+    size_t ShardingKeyLength;
```

Comment:
We prefer use just int for numeric values without special semantics. And in config you should add condition `.GreaterThan(0)`